Std/Lean/Syntax.lean

```diff
@@ -0,0 +1,29 @@
+/-
+Copyright (c) 2022 Microsoft Corporation. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Gabriel Ebner
+-/
+import Lean
+
+/-!
+# Helper functions for working with typed syntaxes.
+-/
+
+namespace Lean
+
+/--
+Applies the given function to every subsyntax.
+
+Like `Syntax.replaceM` but for typed syntax.
```

Comment:
This should mention that it doesn't ensure type-correctness

Reply:
```suggestion
Like `Syntax.replaceM` but for typed syntax. (Note there are no guarantees of type correctness here.)
```